This patch eliminates a segmentation violation with the local RPC
transport method.

--- zrepl-0.5.0/rpc/grpcclientidentity/authlistener_grpc_adaptor.go-orig	Sun Jan  9 05:51:00 2022
+++ zrepl-0.5.0/rpc/grpcclientidentity/authlistener_grpc_adaptor.go	Fri Jul  8 17:23:37 2022
@@ -118,7 +118,11 @@
 		if !ok {
 			panic("peer.FromContext expected to return a peer in grpc.UnaryServerInterceptor")
 		}
-		logger.WithField("peer_addr", p.Addr.String()).Debug("peer addr")
+		peerAddr := ""
+		if p.Addr != nil { // https://github.com/zrepl/zrepl/issues/598
+			peerAddr = p.Addr.String()
+		}
+		logger.WithField("peer_addr", peerAddr).Debug("peer addr")
 		a, ok := p.AuthInfo.(*authConnAuthType)
 		if !ok {
 			panic(fmt.Sprintf("NewInterceptors must be used in combination with grpc.NewTransportCredentials, but got auth type %T", p.AuthInfo))
